test/JDBC/expected/temp_table_rollback_subxact-vu-verify.out

```diff
@@ -0,0 +1,802 @@
+                SELECT * FROM #t1
+            ROLLBACK TRANSACTION T4
+            SELECT * FROM #t1
+        ROLLBACK TRANSACTION T3
+        SELECT * FROM #t1
+    ROLLBACK TRANSACTION T2
+    SELECT * FROM #t1
+ROLLBACK TRANSACTION T1
+SELECT * FROM #t1
+GO
+~~ROW COUNT: 1~~
+
+~~ROW COUNT: 1~~
+
+~~ROW COUNT: 1~~
+
+~~ROW COUNT: 1~~
+
+~~START~~
+varchar
+t1
+t2
+t3
+t4
+~~END~~
+
+~~START~~
+varchar
+t1
+t2
+t3
+~~END~~
+
+~~START~~
+varchar
+t1
+t2
+~~END~~
+
+~~START~~
+varchar
+t1
+~~END~~
+
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: relation "#t1" does not exist)~~
+
+
+SELECT * FROM enr_view;
+GO
+~~START~~
+text
+~~END~~
+
+
+BEGIN TRANSACTION T1
+    SAVE TRANSACTION S2
+        create table #t3(a int)
+        create table #t4(a int identity primary key, b varchar(50))
+        insert into #t3 values (5)
+        insert into #t4 values ('six')
+        SELECT * FROM #t3
+        SELECT * FROM #t4
+        SELECT * FROM enr_view;
+    ROLLBACK TRANSACTION S2
+    SELECT * FROM enr_view;
+COMMIT
+GO
+~~ROW COUNT: 1~~
+
+~~ROW COUNT: 1~~
+
+~~START~~
+int
+5
+~~END~~
+
+~~START~~
+int#!#varchar
+1#!#six
+~~END~~
+
+~~START~~
+text
+#t3
+#t4_a_seq
+#t4
+#pg_toast_#oid_masked#
+#pg_toast_#oid_masked#_index
+#t4_pkey
+~~END~~
+
+~~START~~
+text
+~~END~~
+
+
+BEGIN TRANSACTION T1
+    CREATE TABLE #t1(a int)
+    SAVE TRANSACTION S1
+        DROP TABLE #t1
+        CREATE TABLE #t2(a int)
+        SAVE TRANSACTION S2
+            DROP TABLE #t2
+            CREATE TABLE #t3(a int)
+            SAVE TRANSACTION S3
+            DROP TABLE #t3
+    ROLLBACK TRANSACTION S1
+COMMIT
+GO
+
+-- Should be just #t1. 
+SELECT * FROM enr_view;
+GO
+~~START~~
+text
+#t1
+~~END~~
+
+
+DROP TABLE #t1
+GO
+
+BEGIN TRANSACTION T1
+    CREATE TABLE #t1(a int)
+    SAVE TRANSACTION S1
+        DROP TABLE #t1
+        CREATE TABLE #t2(a int)
+        SAVE TRANSACTION S2
+            DROP TABLE #t2
+            CREATE TABLE #t3(a int)
+            SAVE TRANSACTION S3
+            DROP TABLE #t3
+    ROLLBACK TRANSACTION S1
+ROLLBACK
+GO
+
+-- Should be empty. 
+SELECT * FROM enr_view;
+GO
+~~START~~
+text
+~~END~~
+
+
+CREATE TABLE #t1(a int)
+GO
+BEGIN TRANSACTION T1
+    SAVE TRANSACTION s1
+        DROP TABLE #t1
+        CREATE TABLE #t1(a int identity primary key)  
+        SAVE TRANSACTION s2
+            DROP TABLE #t1
+            CREATE TABLE #t1(a varchar(1000))
+            SAVE TRANSACTION S3
+                DROP TABLE #t1
+    ROLLBACK TRANSACTION s1
+COMMIT
+GO
+
+SELECT * FROM #t1
+GO
+~~START~~
+int
+~~END~~
+
+
+DROP TABLE #t1
+GO
+
+----------------------------------------------------------
+-- General Subtransaction tests
+----------------------------------------------------------
+BEGIN TRANSACTION T1
+    CREATE TABLE #t1(a int primary key, b varchar(16))
+    SAVE TRANSACTION T2
+        INSERT INTO #t1 VALUES (1, 't2')
+        SAVE TRANSACTION T3
+            UPDATE #t1 SET a = 2 WHERE a = 1
```

Comment:
why we have focused a lot on create/drop table testing (which is good) but hardly any coverage on DML side for update and delete?
We should have equally good coverage for DMLs and mix of DDL/DMLs.
Also, let us mix regular tables, temp table and table variables together
Let us add some tests with trigger
Let us add some tests with function calls with select
Let us add some tests with temp tables which does not go to ENR mixed with what goes to ENR
Let us add some tests with sequence of commits followed by a rollback
Let us add some tests with cursors (for base tables at-least with temp table mixed)
Though isolation level should not impact, let us try each isolation level with basic tests
Let us add some tests with inter-operability use cases where PG/TSQL procs are called alternately with transaction commands in both of them
Have we done concurrency tests with this change?

Reply:
I have added the tests as requested, with two notes. 

1. Not sure what you meant by function calls with select. I have added some function calls, but inside the function themselves can't manipulate temp tables in because of TSQL rules. 
2. Triggers aren't allowed on temp tables. There's currently a bug on OSS that allows this (BABEL-4945) but it is fixed correctly in GitFarm. So I did not add any tests for that.

Reply:
1. Use functions with table variables and base tables
2. Use triggers on base tables, see if we can manipulate temp tables and table variables inside trigger procedure
3. what about serializable?
4. Do not see tests related to delete

Reply:
Serializable is not supported. I will add tests for the rest